zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -492,80 +470,95 @@ mod tests {
-            GROUP BY sent_notes.tx, sent_notes.from_account",
+                       0                             AS memo_present
+                FROM   received_notes
+                WHERE  received_notes.spent IS NOT NULL
+            ),
+            sent_note_counts AS (
+                SELECT sent_notes.from_account AS account_id,
+                       sent_notes.tx AS id_tx,
+                       COUNT(DISTINCT sent_notes.id_note) as sent_notes,
+                       SUM(
+                         CASE
+                             WHEN sent_notes.memo IS NULL THEN 0
+                             ELSE 1
+                         END
+                       ) AS memo_count
+                FROM sent_notes
+                LEFT JOIN received_notes
+                          ON (sent_notes.tx, sent_notes.output_pool, sent_notes.output_index) =
+                             (received_notes.tx, 2, received_notes.output_index)
+                WHERE  received_notes.is_change IS NULL
+                   OR  received_notes.is_change = 0
+                GROUP BY account_id, id_tx
+            ),
+            blocks_max_height AS (
+                SELECT MAX(blocks.height) as max_height FROM blocks
+            )
+            SELECT notes.account_id                  AS account_id,
+                   transactions.id_tx                AS id_tx,
+                   transactions.block                AS mined_height,
+                   transactions.tx_index             AS tx_index,
+                   transactions.txid                 AS txid,
+                   transactions.expiry_height        AS expiry_height,
+                   transactions.raw                  AS raw,
+                   SUM(notes.value)                  AS net_transfer,
+                   transactions.fee                  AS fee_paid,
+                   SUM(notes.is_change) > 0          AS has_change,
+                   MAX(COALESCE(sent_note_counts.sent_notes, 0))  AS sent_note_count,
+                   SUM(notes.received_count)         AS received_note_count,
+                   SUM(notes.memo_present) + MAX(COALESCE(sent_note_counts.memo_count, 0)) AS memo_count,
+                   blocks.time                       AS block_time,
+                   (
+                        blocks.height IS NULL
+                        AND transactions.expiry_height <= blocks_max_height.max_height
+                   ) AS expired_unmined
+            FROM transactions
+            JOIN notes ON notes.id_tx = transactions.id_tx
+            JOIN blocks_max_height
+            LEFT JOIN blocks ON blocks.height = transactions.block
+            LEFT JOIN sent_note_counts
+                      ON sent_note_counts.account_id = notes.account_id
+                      AND sent_note_counts.id_tx = notes.id_tx
+            GROUP BY notes.account_id, transactions.id_tx",
+            // v_tx_events
+            "CREATE VIEW v_tx_events AS
```

Comment:
Is this the "Wallet Events" details that we discussed on our last sync?

I assume that the `v_transactions` is the `ZcashTransaction.Overview` and then there would be a drill down to details where the wallet requests all events for a given `id_tx` (the internal id in the database present on the Overview entity) and it would be able to "enrich" the Overview information with the wallet events. (we could make a `ZcashTransaction.Detail` entity to reflect this)

Reply:
This is effectively (and maybe should be named) `v_tx_outputs`. It's the list of outputs of the transaction that are associated with this wallet: all outputs received (with change marked) and outputs sent.

Reply:
To determine whether an output is "sent" or "received" from the perspective of a given account, a wallet should consider the value sof the `from_account`, `to_account`, and `to_address` fields. Either `from_account`, `to_account`, or both will always be populated; there should never be an occasion where both are null. The valid combinations are:

(from_account, to_account, null)
(from_account, null, null)
(from_account, null, to_address)
(null, to_account, null)